cpp/src/parquet/column_writer.cc

```diff
@@ -460,7 +460,11 @@ class SerializedPageWriter : public PageWriter {
         ToThrift(page.definition_level_encoding()));
     data_page_header.__set_repetition_level_encoding(
         ToThrift(page.repetition_level_encoding()));
-    data_page_header.__set_statistics(ToThrift(page.statistics()));
+
+    // Write page statistics only when page index is not enabled.
```

Comment:
Seems that this will invalidate page filter?

Reply:
Yes, in this case it is more efficient to use the column index instead.